pkg/resources/templates/config/nifi_properties.go

```diff
@@ -217,11 +309,26 @@ nifi.cluster.firewall.file=
 nifi.cluster.flow.election.max.wait.time=1 mins
 nifi.cluster.flow.election.max.candidates=
 
+# cluster load balancing properties #
+nifi.cluster.load.balance.host=
+nifi.cluster.load.balance.port=6342
+nifi.cluster.load.balance.connections.per.node=1
+nifi.cluster.load.balance.max.thread.count=8
+nifi.cluster.load.balance.comms.timeout=30 sec
+
 # zookeeper properties, used for cluster management #
 nifi.zookeeper.connect.string= {{ .ZookeeperConnectString }}
 nifi.zookeeper.connect.timeout=3 secs
 nifi.zookeeper.session.timeout=3 secs
 nifi.zookeeper.root.node={{ .ZookeeperPath }}
+nifi.zookeeper.client.secure=false
+nifi.zookeeper.security.keystore=
+nifi.zookeeper.security.keystoreType=
+nifi.zookeeper.security.keystorePasswd=
+nifi.zookeeper.security.truststore=
+nifi.zookeeper.security.truststoreType=
+nifi.zookeeper.security.truststorePasswd=
+nifi.zookeeper.jute.maxbuffer=
```

Comment:
```suggestion
# TODO: implement exclusive definition with kubernetes implementation
nifi.zookeeper.connect.string= {{ .ZookeeperConnectString }}
nifi.zookeeper.connect.timeout=3 secs
nifi.zookeeper.session.timeout=3 secs
nifi.zookeeper.root.node={{ .ZookeeperPath }}
nifi.zookeeper.client.secure=false
nifi.zookeeper.security.keystore=
nifi.zookeeper.security.keystoreType=
nifi.zookeeper.security.keystorePasswd=
nifi.zookeeper.security.truststore=
nifi.zookeeper.security.truststoreType=
nifi.zookeeper.security.truststorePasswd=
nifi.zookeeper.jute.maxbuffer=
```

Reply:
I'm not sure we should. To migrate the cluster states from ZooKeeper to `ConfigMaps`, we need to be able to specify the ZooKeeper configuration and configure Kubernetes as Cluster Manager.

Reply:
I'm not sure it's something the operator should do for the user -- I just thought it was worth bringing up!

Here's where it was added: https://issues.apache.org/jira/browse/NIFI-10976

Reply:
No but we should be able to set the properties through the operator for ZooKeeper no?

The user still need to set `nifi.state.management.provider.cluster.previous=zk-provider`.
But the operator will set
```
nifi.state.management.provider.cluster=kubernetes-provider
nifi.zookeeper.connect.string= {{ .ZookeeperConnectString }}
nifi.zookeeper.connect.timeout=3 secs
nifi.zookeeper.session.timeout=3 secs
nifi.zookeeper.root.node={{ .ZookeeperPath }}
nifi.zookeeper.client.secure=false
nifi.zookeeper.security.keystore=
nifi.zookeeper.security.keystoreType=
nifi.zookeeper.security.keystorePasswd=
nifi.zookeeper.security.truststore=
nifi.zookeeper.security.truststoreType=
nifi.zookeeper.security.truststorePasswd=
nifi.zookeeper.jute.maxbuffer=
```
no?

If the operator don't set the ZooKeeper properties, the user will have to.

Reply:
Yes, the operator should configure zookeeper or kubernetes state providers, but it should not perform a state migration from zookeeper to kubernetes. Do you disagree?

Reply:
No I agree. But if we say that the operator configures either Kubernetes or ZooKeeper but not both of them at the same time.  The user won't be able to perform a state migration easly. No?

Reply:
It actually looks like it's handled automatically if you set the `nifi.state.management.provider.cluster.previous` property, for example to `zk-provider`. But as you say, the zookeeper configuration needs to be there as well.

I think given this, it's possible for the user to provide all the necessary configuration via `nifi.properties` to migrate from zookeeper to kubernetes _as long as_ nifikop allows the user to configure both zk and k8s state providers (only setting 1 to the active one).

Reply:
So the ZooKeeper properties should be only set if the `ClusterManager` is set on it. And in case of migration, you will to set them manually by overring the `nifi.properties`. It's good to me, just to make sure it's what we want.

Reply:
Yes, looking though all of the changes here and how the config gets templated by nifikop, it should be possible to automatically migrate from an existing zookeeper to kubernetes by setting `nifi.state.management.provider.cluster.previous=zk-provider`, providing all necessary zookeeper configuration in `NifiCluster` spec and in `nifi.properties` (basically leave all previous config in place), and setting `clusterManager=kubernetes` in `NifiCluster` spec.

Reply:
So the ZooKeeper should not be conditionned to `ClusterManager=zookeeper`, right?

Reply:
Correct. It should not be. It should be written to the nifi.properties if its in the NifiCluster spec always.